iOS/MasterTimeline/MasterTimelineViewController.swift

```diff
@@ -887,6 +898,25 @@ private extension MasterTimelineViewController {
 		}
 		return action
 	}
+	
+	func copyArticleURLAction(_ article: Article) -> UIAction? {
+		guard let preferredLink = article.preferredLink, let url = URL(string: preferredLink) else { return nil }
```

Comment:
This pull request is against main (which is indeed what the contributing guidelines say to do) rather than against the two branches where recent commits have been pulled (ios-candidate and mac-release), and so it hasn't picked up on the addition of article.preferredURL—which definitely would be used here instead of line 903, and possibly one or two other places in the pull request. I'd suggest the pull request probably needs to be rebased on those recent changes? Otherwise this patch re-introduces the bugs fixed when preferredLink strings are malformed in the source feed.

Reply:
Actually let me correct the above. It doesn't re-introduce the bug per se, it is just vulnerable to the issue which means occasionally it will fail to be able to share a link that otherwise would have been shared. All other places this has been fixed in the code base but this would become anew one not fixed.

Reply:
We have since back merged the changes in `mac-release` and `ios-candidate` to `main`. This means that `Artilcle.preferredURL` is now on `main` and @babbage 's suggest change can be made.

Reply:
Thanks for the update. I'll rebase and adjust to use `preferredURL`